deps.js

```diff
@@ -0,0 +1,201 @@
+const {curry, uniq} = require(`lodash`);
+const path = require(`path`);
+const rpj = require(`read-package-json`);
+
+const readPackage = denodeify(rpj);
+
+function findDepsForPackage(packagePath) {
+  return readPackage(packagePath)
+   .then((pkg) => {
+     if (pkg.main) {
+       return walkDeps(path.resolve(path.dirname(packagePath), pkg.main));
+     }
+     else if (pkg.bin) {
+       return Object.values(pkg.bin).reduce((acc, bin) => acc.concat(walkDeps(path.resolve(path.dirname(packagePath), bin))), []);
+     }
+
+     try {
+       const p = path.resolve(path.dirname(packagePath), `index.js`);
+       fs.statSync(p);
+       return walkDeps(p);
+     }
+     catch (err) {
+       if (err.code === `ENOENT`) {
+         throw new Error(`cannot determine entrypoint for ${pkg.name}`);
+       }
+       throw err;
+     }
+   })
+   .catch((err) => {
+     if (err.code === `EISDIR`) {
+       return findDepsForPackage(path.join(packagePath, `package.json`));
+     }
+
+     throw err;
+   });
+}
+
+const depsToVersions = curry((rootPkg, deps) => deps.reduce((acc, dep) => {
+  if (builtins.includes(dep)) {
+    return acc;
+  }
+  acc[dep] = rootPkg.dependencies[dep] || rootPkg.devDependencies[dep] || rootPkg.optionalDependencies[dep];
+  if (!acc[dep]) {
+    try {
+      acc[dep] = require(`./packages/node_modules/${dep}/package.json`).version;
+    }
+    catch (err) {
+      // eslint-disable-next-line no-console
+      console.warn(err);
+      throw new Error(`Failed to determine version for ${dep}, Is it missing from package.json?`);
+    }
+  }
+  return acc;
+}, {}));
+
+const assignDeps = curry((pkgPath, deps) => {
+  if (!pkgPath.startsWith(`./`) && !pkgPath.startsWith(`/`)) {
+    pkgPath = `./${pkgPath}`;
+  }
+  const pkg = require(pkgPath);
+  pkg.dependencies = deps;
+  fs.writeFileSync(pkgPath, `${JSON.stringify(pkg, null, 2)}\n`);
+});
+
+const visited = new Set();
+function walkDeps(filePath) {
+  if (visited.has(filePath)) {
+    return [];
+  }
+  try {
+    visited.add(filePath);
+    const requires = detective(fs.readFileSync(filePath));
+    const deps = requires.reduce((acc, dep) => {
+      if (dep.startsWith(`.`)) {
+        acc = acc.concat(walkDeps(path.resolve(path.dirname(filePath), dep)));
+      }
+      else {
+        acc.push(dep);
+      }
+      return acc;
+    }, []);
+    return uniq(deps.map((d) => {
+      d = d.split(`/`);
+      if (d[0].startsWith(`@`)) {
+        return d.slice(0, 2).join(`/`);
+      }
+      return d[0];
+    }));
+  }
+  catch (err) {
+    if (err.code === `EISDIR`) {
+      return walkDeps(path.resolve(filePath, `index.js`));
+    }
+    if (err.code === `ENOENT` && !filePath.endsWith(`.js`)) {
+      return walkDeps(`${filePath}.js`);
+    }
+    throw err;
+  }
+}
+
+function updateSinglePackage(packagePath) {
+  return findDepsForPackage(packagePath)
+    .then(depsToVersions(require(`./package.json`)))
+    .then(assignDeps(path.join(packagePath, `package.json`)));
```

Comment:
nope. behold the power of `curry`. `assignDeps` is a curried function, so, when it's called at line 125, it returns a function that'll be passed to `.then` expecting the other parameter.